src/bees.cc

```diff
@@ -597,6 +599,8 @@ bees_main(int argc, char *argv[])
 
 	THROW_CHECK1(invalid_argument, argc, argc >= 0);
 
+	string cwd(readlink_or_die("/proc/self/cwd"));
```

Comment:
Wouldn't it make more sense to use `name_fd(root_fd())` instead of cwd here (well, not here, but in BeesContext's constructor)?

cwd isn't used by bees anywhere else (except as a default location for core dumps if the user is not running something that overrides the default kernel core dumper).  Relative paths within bees are always relative to some other file descriptor.

The root of the filesystem seems like the logical origin point for relative paths; however, we could have one option which is the name of a directory to which all paths will be made relative (with the special case that using "/" as this directory gives absolute paths).  That would allow users to have something like '/var/lib/bees/mnt/foo/bar' be reduced to 'foo/bar' if bees was run with 'bees -p /var/lib/bees/mnt /var/lib/bees/mnt/foo/bar'.

Reply:
I found no way to access the root fd from the chatter... That's why I chose that location and gone that way.

Reply:
I was thinking you would do the call to set_relative_path() from the point where root_fd() is set.

Reply:
But still I have no easy access to the context from the chatter without changing interfaces all over. I tried to modify the chatter constructor to store the context but there are places where it simply doesn't make sense. And name_fd() is a helper which absolutely has no knowledge about the context. And I didn't want to touch all of the chatter invocation all over the code so I decided to simply modify name_fd() (as it seems to be solely used to generate log output) and enable it to make paths relative to the cwd of the main process which can be easily modified.

I was also thinking about having it relative per each context. But this makes log output harder to read: If you give bees more than one fs to work on, you wouldn't be able to easily decipher which fs the log belongs to. But I could, as admin starting the service, easily cwd to the most common parent directory and it will just work, e.g. everything can start from /run/bees (when using the systemd starter) and using the cwd, I would easily see the uuid per each log entry.

I think the latter is more useful although I'm with you that the current solution doesn't look very nice and elaborated. We could add an optional argument to "--relative-path" to accept what relative root you want to use instead of relying solely on cwd.

Reply:
BTW: When I was thinking again about it later, I also found that "--absolute-path" is probably the wrong wording as it doesn't really make absolute paths, it just doesn't cut away strings from the front. You could still give bees a relative path on the cmdline.

Reply:
More specifically, I was thinking of something like:

```
diff --git a/src/bees-context.cc b/src/bees-context.cc
index 4881fa70..7cd34570 100644
--- a/src/bees-context.cc
+++ b/src/bees-context.cc
@@ -828,6 +828,12 @@ BeesContext::set_root_fd(Fd fd)
                return resolve_addr_uncached(addr);
        });
 
+       // Look at options saved by main()
+       if (use_relative_paths) {
+               crucible::set_relative_path("");
+               crucible::set_relative_path(name_fd(fd));
+       }
+
```

i.e. let BeesContext get the configuration information through global variables (or static members of BeesContext) instead of trying to make Chatter understand a BeesContext. 

An even better way would be to just extend BeesContext::set_root_path, or add a new method that just sets up name_fd().

Reply:
Hmm... That's an idea... I'll look into it. Could we first merge some of the previous PRs if they look good to you so I could rebase my work upon a new master having the basics?

Reply:
I have a couple of them merged already locally, hang on...